### ISSUE TITLE
Centralize quality checks

### ISSUE THREAD

**Issue by pykong** [pykong/copier#132]
Reminder: Think about best way to have all quality checks in a single place.
Steps:
- formatting (**black**)
- linting (**flake8** and its plugins)
- testing (**pytest**)
- static analysis (**mypy**)

Currently
- `ci.yml`
- `.pre-commit-config.yaml`
and to a lesser degree 
- `pyproject.toml`
- `Makefile`

All compete for this role.

**Comment by Yajo** [comment]
IMHO:

- formatting & linting: pre-commit
- testing & static analysis: CI
- entrypoint for dev/lint/test purposes: My vote still goes for invoke, I'm having a good experience on it.

However, note that pre-commit also can use `stage: manual` which does additional tests on manual trigger. https://pre-commit.com/#confining-hooks-to-run-at-certain-stages

IMHO the CI should just leverage pre-commit or invoke. If you want some inspiration on invoke, go check the `tasks.py` file I'm developing in https://github.com/Tecnativa/doodba-scaffolding/pull/69/files (VERY WIP, will change).

I like invoke in the sense that it vendors all its dependencies (so there will be no dependency conflicts with any other tool in your belt), and it can be used even to install things such as poetry and pre-commit.

Well, any solution will be good in any case :blush:

**Comment by pykong** [comment]
- formatting & linting: pre-commit
- testing & static analysis: CI

That is basically how I have implemented it currently.

We might still use `invoke` to provide defined entry points for "test" and "lint".

So for me the question boils down whether we need additional dev tooling in the form of `invoke` at all. This is to be discussed here: https://github.com/pykong/copier/issues/129#issuecomment-592515765

Closing as the central question has been answered.
